src/model/transaction/__tests__/moveBlockInContentState-test.js

```diff
@@ -0,0 +1,150 @@
+/**
+ * Copyright (c) 2013-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @emails oncall+ui_infra
+ * @format
+ */
+
+'use strict';
+
+jest.disableAutomock();
+
+jest.mock('generateRandomKey');
+
+const EditorState = require('EditorState');
+const ContentBlock = require('ContentBlock');
+const ContentBlockNode = require('ContentBlockNode');
+const ContentState = require('ContentState');
+const Immutable = require('immutable');
+
+const moveBlockInContentState = require('moveBlockInContentState');
+
+const {List} = Immutable;
+
+const contentBlocks = [
+  new ContentBlock({
+    key: 'A',
+    text: 'Alpha',
+  }),
+  new ContentBlock({
+    key: 'B',
+    text: 'Beta',
+  }),
+  new ContentBlock({
+    key: 'C',
+    text: 'Charlie',
+  }),
+];
+
+const contentBlockNodes = [
+  new ContentBlockNode({
+    key: 'A',
+    text: 'Alpha',
+    nextSibling: 'B',
+  }),
+  new ContentBlockNode({
+    key: 'B',
+    text: '',
+    children: List(['C']),
+    nextSibling: 'D',
+    prevSibling: 'A',
+  }),
+  new ContentBlockNode({
+    key: 'C',
+    parent: 'B',
+    text: 'Charlie',
+  }),
+  new ContentBlockNode({
+    key: 'D',
+    text: '',
+    prevSibling: 'B',
+    children: List(['E']),
+  }),
+  new ContentBlockNode({
+    key: 'E',
+    parent: 'D',
+    text: 'Elefant',
+  }),
+];
+
+const assertMoveBlockInContentState = (
+  blockToBeMovedKey,
+  targetBlockKey,
+  insertionMode,
+  blocksArray = contentBlocks,
+) => {
+  const editor = EditorState.createWithContent(
+    ContentState.createFromBlockArray(blocksArray),
+  );
+  const contentState = editor.getCurrentContent();
+  const blockToBeMoved = contentState.getBlockForKey(blockToBeMovedKey);
+  const targetBlock = contentState.getBlockForKey(targetBlockKey);
+
+  expect(
+    moveBlockInContentState(
+      contentState,
+      blockToBeMoved,
+      targetBlock,
+      insertionMode,
+    )
+      .getBlockMap()
+      .toSetSeq()
+      .toArray()
+      .map(filter => {
+          const {
+             data,
+             characterList,
+             depth,
+             type,
+             text,
+             ...other,
+          } = filter.toJS();
+        return other;
+      }),
+  ).toMatchSnapshot();
+};
+
+test('must be able to move block before other block', () => {
+  assertMoveBlockInContentState('C', 'A', 'before');
+});
+
+test('must be able to move block after other block', () => {
+  assertMoveBlockInContentState('A', 'C', 'after');
+});
+
+test('must be able to move nested block before other block', () => {
```

Comment:
I'm missing something here - in the snapshot it looks like it moved D and E too, even though they are not children of C.

Reply:
You are correct, just checked on the snapshot and it is a bug! will fix this! thank you for spotting this!

Reply:
Turns out this is a bug in immutable where they do not recognize `null` as a false value for `takeWhile` just checked with lee and will submit a separate PR for immutable. In the meantime we will just cast that to a boolean with `!!(...)`